### PR TITLE
Add options to cli in place of environment variables

### PR DESCRIPTION
**Pull request by kbenne**

### CODE REVIEW

File: bin/openstudio_meta [hunk 1]
```diff
@@ -23,9 +23,10 @@ Signal.trap('INT') { abort }
 require 'logger'
 $logger = ::Logger.new(STDOUT)
 $logger.level = ::Logger::WARN
-ENV['GEM_PATH'] = File.absolute_path(File.join(__FILE__, './../../gems/')).to_s + File::PATH_SEPARATOR + File.absolute_path(File.join(__FILE__, './../../gems/bundler/gems')).to_s
 ENV['GEM_HOME'] = File.absolute_path(File.join(__FILE__, './../../gems/'))
 
+$ruby_path = "#{RbConfig::CONFIG['prefix']}/bin/ruby"
```
Comment (rHorsey) [inline bin/openstudio_meta]: So much better - thanks!

File: bin/openstudio_meta [hunk 2]
```diff
@@ -23,9 +23,10 @@ Signal.trap('INT') { abort }
 require 'logger'
 $logger = ::Logger.new(STDOUT)
 $logger.level = ::Logger::WARN
-ENV['GEM_PATH'] = File.absolute_path(File.join(__FILE__, './../../gems/')).to_s + File::PATH_SEPARATOR + File.absolute_path(File.join(__FILE__, './../../gems/bundler/gems')).to_s
```
Comment (rHorsey) [inline bin/openstudio_meta]: You're dealing with this elsewhere in the packaging?

File: bin/openstudio_meta [hunk 3]
```diff
@@ -62,6 +74,28 @@ def parse_options(opts = nil, argv = nil)
...
+    FileUtils.remove_entry ENV['OS_SERVER_LOG_PATH']
+    ENV['OS_SERVER_LOG_PATH'] = path
+  end
+
+  ENV['RAILS_ENV'] = 'local'
+  opts.on('--rails-env [ENV]', 'Rails Environment') do |env|
+    ENV['RAILS_ENV'] = env
+  end
+
+  ENV['OS_SERVER_HOST_URL'] = 'http://localhost:8080'
```
Comment (kbenne) [inline bin/openstudio_meta]: I would prefer to just let that dynamic thing do its work, but from empirical testing I found that if OS_SERVER_HOST_URL is not set in the environment then the install gems command fails. 

Granted I deduced that before you made you your last commit, so things might have changed.

If the port is selected randomly, how does PAT find out where to connect?

File: bin/openstudio_meta [hunk 4]
```diff
@@ -326,10 +362,16 @@ class StartLocal
   # @return [Fixnum] Return status
   #
   def execute(sub_argv)
-    options = { debug: false, force_directories: false, workers: 1 }
+    mongod = which 'mongod'
```
Comment (kbenne) [inline bin/openstudio_meta]: eh. it is just a convenience. if mongo is in the path and a mongo option was not passed, then just use what you find in the path.

PAT will obsoletely pass an explicit path to mongo. I just got tired of typing out all of theses paths in testing.

File: bin/openstudio_meta (hunk 2 above)
Comment (kbenne) [inline bin/openstudio_meta]: I honestly don't know the different between GEM_PATH and GEM_HOME, but in practice it seems that as along as GEM_PATH is set, everything seems to work.

File: bin/openstudio_meta (hunk 4 above)
Comment (rHorsey) [inline bin/openstudio_meta]: Sooooooo.... Thank you!!! ;)

**Comment by rHorsey** [comment]
I'm comfortable with merging this. I still think the GEM_PATH thing may bite us later, but if so it's an obvious error and a trivial fix. And if the $ruby_path changes work cross-platform then I'd prefer that regardless.

**Pull request merged**